src/App.tsx

```diff
@@ -1,61 +1,150 @@
+        action="/api/users"
+        method="POST"
+        onSubmit={addTodo}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            {' '}
+            <input
+              name="title"
+              value={todoState.title}
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              onChange={(event) => {
+                handleChange(event);
+              }}
+            />
+          </label>
+
+          {titleError
+          && (<span className="error">Please enter a title</span>)}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            User:
+            {' '}
+            <select
+              data-cy="userSelect"
+              value={todoState.select}
+              name="select"
+              onChange={(event) => {
+                handleChange(event);
+              }}
+            >
+              <option
+                value="0"
+                disabled
+              >
+                Choose a user
+              </option>
+
+              {[...usersFromServer].map(user => (
+                <option
+                  id={`${user.id}`}
+                  value={user.name}
+                  data-username={user.username}
+                  data-email={user.email}
+                >
+                  {user.name}
+                </option>
+              ))}
+            </select>
+          </label>
+
+          {selectError
+          && (<span className="error">Please choose a user</span>)}
 
-          <span className="error">Please choose a user</span>
         </div>
 
         <button type="submit" data-cy="submitButton">
           Add
         </button>
       </form>
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoList todos={todoList} userList={usersFromServer} />
```

Comment:
There is no need to pass users to TodoList. You can prepare todos and add user property for each todo
![image](https://user-images.githubusercontent.com/96409563/229041496-09092012-9e60-49c1-b757-e2b8c31435d3.png)